runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -2248,7 +2248,7 @@ MM_CopyForwardScheme::updateMarkMapAndCardTableOnCopy(MM_EnvironmentVLHGC *env,
 				/* do nothing */
 				break;
 			default:
-				Assert_MM_unreachable();
+				Assert_GC_true_with_message4(env, false, "Unexpected Card state %zu for Card %p, srcObject %p, dstObject %p\n", *card, card, srcObject, dstObject);
```

Comment:
Is `%zu` correct? It seems `Card` is U_8.

Reply:
correct, Card is U_8. Do you see any problem use `%zu`? What do you think would fit better? `%d` ? `%z`?

Reply:
`%zu` is 32-bits on a 32-bit JVM and 64-bits on a 64-bit JVM, which doesn't apply to U_8. I believe using `%u` is correct.

Reply:
fyi https://github.com/eclipse-openj9/openj9/blob/master/doc/diagnostics/AddingTracepoints.md
'%zu` "platform-dependent size_t (equivalent to J9 UDATA), formatted as unsigned decimal"